postgres-native-sqldelight-driver/src/commonMain/kotlin/app/softwork/sqldelight/postgresdriver/PostgresNativeDriver.kt

```diff
@@ -111,7 +111,7 @@ public class PostgresNativeDriver(private var conn: CPointer<PGconn>) : SqlDrive
         parameters: Int,
         binders: (SqlPreparedStatement.() -> Unit)?
     ): QueryResult.Value<R> {
-        val cursorName = if (identifier == null) "myCursor" else "cursor$identifier"
+        val cursorName = if (identifier == null) "myCursor" else "cursor${identifier.toString().replace("-", "_")}"
```

Comment:
I think the only downside here is that `replace` loops over all the `Char` in the `String`, perhaps a bit micro-optimisation but the function below is simply enough without complicating the code a lot.

```kotlin
private fun Int.asCursorIdentifier(): String  =
    if(this < 0) "_${toString().substring(1)}" else toString()
```

Changing the snippet to:

`"cursor${identifier.asCursorIdentifier()}"`

Reply:
thanks, was unaware of the loop!